src/android/camera_hal_config.cpp

```diff
@@ -357,15 +357,8 @@ CameraHalConfig::CameraHalConfig()
  */
 int CameraHalConfig::parseConfigurationFile()
 {
-	std::filesystem::path filePath = LIBCAMERA_SYSCONF_DIR;
```

Comment:
It's a bit unfortunate not to be able to use the standard library - but I guess that's a limitation on Android.
And given that this is an 'android hal' it's probably not unreasonable to highglight that as an issue.
We use the HAL in ChromeOS where we can utilise the std::filesystem.
I wonder if it can be wrapped in some way so we can take advantage of the std::filesystem when it's there or not otherwise.

Reply:
std::filesystem is present in AOSP but not available for vendor libraries.
To make it available `Android.bp` should specify `vendor: true` field.
But for some reason nobody added it there.

It is a single usage. Do we really want to have 2 implementations here depending on presence of std::filesystem?